internal/release/render/render.go

```diff
@@ -162,6 +163,13 @@ func getReleaseViaPrompt(releases []*cross.Release, env string) (*v1alpha1.Relea
 func hydrateValues(release *v1alpha1.Release, environment *v1alpha1.Environment) (map[string]any, error) {
 	values := release.Spec.Values
 
+	// --- This replicates the behavior we have encoded into the application set.
+	if release.Spec.Version != "" {
+		setInMap(values, []string{"image", "tag"}, release.Spec.Version)
+	}
+	setInMap(values, []string{"common", "annotations", "nesto.ca/deployed-by"}, "joy")
+	// --- end of appset behaviour
+
```

Comment:
Hmmmm, this is starting to drift into having nesto specific logic in joy, which i'm not sure that we want...

Reply:
This is true. But this is because we've coupled our appset to our generic chart and we cannot do release rendering in a sane way without knowing these.

The simplest would have been to not have the release.spec.version coupled to the image tag.
Also it would have been good to not have the appset muddy the values of the charts with defaults from the release spec.

That being said, I have been using this command more and more as I work with joy releases. It's actually great when you want to quickly know what a change to a release would produce. The real source of truth is always in argo/k8 but i think getting this to work correctly. Is important especially as we migrate towards joy internally.

My vote is that we allow this much internal logic to bleed into the Joy CLI for now, and work to decouple some stuff so that we can remove this.

@silphid what are your thoughts?

Reply:
As a thought, if a user from outside nesto opens an issue because the release render applies `image.tag` to their chart values in an undesirable way, then we can address it. However the likelihood of that happening is near zero.

Reply:
For `common.annotations.nesto\.ca/deployed\-by: joy` we should do same as for defaulting chart to `generic`, by introducing a catalog-level config for specifying a list of global custom chart parameters. For example:
```
globalChartParams:
  common.annotations.nesto\.ca/deployed\-by: joy
```
(my understanding is that the backslashes are required by helm to properly escape the `nesto.ca/deployed-by` annotation key)

For `image.tag` it's slightly different, as it's just the default version key to use if not specified in release. So I'd simply create a `defaultVersionKey` catalog-level config that we would set to `image.tag` in our case.

Note: Eventually, we should also have the generator use those configs dynamically instead of hard-coding them in the app set, so we have a single source of truth that is used cohesively by both CLI and generator.